relay_messages_test.go

```diff
@@ -524,3 +550,47 @@ func TestLazyErrorCodes(t *testing.T) {
 		assert.Equal(t, ec, f.Code(), "Mismatch between error code and lazy frame's Code() method.")
 	})
 }
+
+// TODO(cinchurge): replace with e.g. decodeThriftHeader once we've resolved the import cycle
+func uint16KeyValToMap(tb testing.TB, buffer []byte) map[string]string {
+	rbuf := typed.NewReadBuffer(buffer)
+	nh := int(rbuf.ReadUint16())
+	retMap := make(map[string]string, nh)
+	for i := 0; i < nh; i++ {
+		keyLen := int(rbuf.ReadUint16())
+		key := rbuf.ReadBytes(keyLen)
+		valLen := int(rbuf.ReadUint16())
+		val := rbuf.ReadBytes(valLen)
+		retMap[string(key)] = string(val)
+	}
+	require.NoError(tb, rbuf.Err())
+	return retMap
+}
+
+func TestLazyCallReqContents(t *testing.T) {
+	cr := reqHasAll.reqWithParams(t, testCallReqParams{
+		arg2Buf: thriftarg2test.BuildKVBuffer(map[string]string{
+			"foo": "bar",
+			"baz": "qux",
+		}),
+		arg3Buf: []byte("some arg3 data"),
```

Comment:
can we move the "some arg3 data" to a constant and use it here and below when validating the value

can do the same for the map of headers too